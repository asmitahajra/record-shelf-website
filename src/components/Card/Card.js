/* eslint-disable react/prop-types */
import './Card.scss';
import React from 'react';
// import { Link } from 'react-router-dom';

const Card = ({ eachSong, onIncrement }) => {
  const {
    nameSong, albumArtUrl, artist, count, like,
  } = eachSong;
  let heart;
  if (like === true) {
    heart = '../assets/heart-red.svg';
  } else {
    heart = '../assets/heart-gray.svg';
  }
  return (
    <div className="card">
      <img data-testid="card-img" src={albumArtUrl} alt={nameSong} className="card-image" />
      <div className="details">
        <div className="song-artist">
          <div className="song">
            {nameSong}
          </div>
          <br />
          <div className="artist">
            <b>{artist}</b>
          </div>
        </div>
        <div className="like">
          <button type="button" onClick={onIncrement}>
            <img data-testid="card-img" src={heart} alt="like-button" className="like-image" />
            {count}
          </button>
        </div>
      </div>
    </div>
  );
};

export default Card;
