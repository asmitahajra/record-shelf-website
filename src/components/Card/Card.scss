// .card {
//     box-shadow: 10px 10px 5px grey;
//     padding: 0.5rem;
//     border: black solid 1px;
//     border-radius: 0.9rem;
//     margin: 0.9rem;
//     width: 14rem;

//     .image {
//         width: 100%;
//         //width: 10rem;
//         height: 15rem;
//     }
// }

// .product_card_light {
//     box-shadow: 10px 10px 5px grey;
//     padding: 0.5rem;
//     border: black solid 1px;
//     border-radius: 0.9rem;
//     margin: 0.9rem;
//     width: 14rem;
// }


.card {

    box-shadow: 10px 10px 5px grey;
    padding: 0.5rem;
    border: black solid 1px;
    border-radius: 0.9rem;
    margin: 0.9rem;
    width: 14rem;
    background-color: #DFDFDF;

    .card-image {
            width: 100%;
            //width: 10rem;
            height: 15rem;
            border-radius: 10rem;
    }

    .details {
        display: flex;
        justify-content: space-between;

        .song-artist {
            display: flex;
            flex-direction: column;
        }

        .like {
            .like-image {
                width: 2rem;
                height: 2rem;
            }
            
        }
    }

    .song {
        font-size: large;
        font-family:Arial, Helvetica, sans-serif
    }

}
