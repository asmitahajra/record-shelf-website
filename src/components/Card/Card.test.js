/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

describe(Card.name, () => {
  const mockProp = {
    onIncrement: jest.fn(),
    eachSong: {
      nameSong: 'artist',
      artist: 'abc',
      count: 5,
      like: true,
      albumArtUrl: 'abc',
    },
  };

  test('should match snapshot', () => {
    const { container } = render(<Card {...mockProp} />);
    expect(container).toMatchSnapshot();
  });
});
