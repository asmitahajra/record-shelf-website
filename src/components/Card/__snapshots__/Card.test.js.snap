// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="artist"
      class="card-image"
      data-testid="card-img"
      src="abc"
    />
    <div
      class="details"
    >
      <div
        class="song-artist"
      >
        <div
          class="song"
        >
          artist
        </div>
        <br />
        <div
          class="artist"
        >
          <b>
            abc
          </b>
        </div>
      </div>
      <div
        class="like"
      >
        <button
          type="button"
        >
          <img
            alt="like-button"
            class="like-image"
            data-testid="card-img"
            src="../assets/heart-red.svg"
          />
          5
        </button>
      </div>
    </div>
  </div>
</div>
`;
